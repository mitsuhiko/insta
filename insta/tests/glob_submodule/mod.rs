#![cfg(feature = "glob")]

#[test]
fn test_basic_globbing_parent_dir() {
    insta::glob!("../inputs", "*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_json_snapshot!(&contents);
    });
}

#[test]
fn test_basic_globbing_nested_parent_dir_base_path() {
    insta::glob!("../inputs-nested", "*/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_snapshot!(&contents);
    });
}

#[test]
fn test_basic_globbing_nested_parent_glob() {
    insta::glob!("..", "inputs-nested/*/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_snapshot!(&contents);
    });
}

#[test]
fn test_globs_follow_links_parent_dir_base_path() {
    insta::glob!("../link-to-inputs", "*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_json_snapshot!(&contents);
    });
}

#[test]
fn test_globs_follow_links_parent_dir_glob() {
    insta::glob!("..", "link-to-inputs/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        insta::assert_json_snapshot!(&contents);
    });
}

#[test]
fn test_basic_globbing_absolute_dir() {
    insta::glob!(
        concat!(env!("CARGO_MANIFEST_DIR"), "/tests/inputs"),
        "*.txt",
        |path| {
            let contents = std::fs::read_to_string(path).unwrap();
            insta::assert_json_snapshot!(&contents);
        }
    );
}
