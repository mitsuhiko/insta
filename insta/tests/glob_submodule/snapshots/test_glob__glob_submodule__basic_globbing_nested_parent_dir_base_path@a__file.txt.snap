---
source: tests/glob_submodule/mod.rs
expression: "&contents"
input_file: tests/inputs-nested/a/file.txt
---
Hello A

